individual/services.py

```diff
@@ -697,7 +697,7 @@ def _validate_location(
         result = {
             'field_name': 'location_name',
         }
-        if (location_name is None or location_name == "") and (location_code is None or location_code == ""):
+        if (pd.isna(location_name) or location_name == "") and (pd.isna(location_code) or location_code == ""):
```

Comment:
as this a service I believe it should not assume it always come from data frame
```suggestion
        if (pd.isna(location_name) or not location_name) and (pd.isna(location_code) or not location_code):
```

Reply:
`_validate_location` is only called from `process_chunk` above where `location_name` (and `location_code`) are `row.location_name` (and `row.location_code`) and row is guaranteed to be from a datafrom: `for _, row in chunk.iterrows():`. Plus, `pd.isna(None)` returns True so it would have taken care of the location_name=None (location_code=None) case anyway, even if it's not from a dataframe. `not location_name` would not work for the case of a dataframe, but the proposed implementation would work for both dataframe or otherwise.